tests/test_app.py

```diff
@@ -416,27 +430,97 @@ def test_only_running_once():
     """
     If we are the first to acquire the application lock we should succeed
     """
+    setup_exception_handler()
     check_only_running_once()
+    _shared_memory.release()
     assert True
 
 
 def test_running_twice():
```

Comment:
Rewritten test that uses chaining of process launching to avoid `sleep(x)` and a timing sensitive test that takes massively different amounts of time on different test runners.